Make.defs

```diff
@@ -13,6 +13,10 @@ OMIT_PAM_MODULE =
 # May be needed when compiling to use with OpenPAM.
 PAM_SO_SUFFIX =
 
+# Flags for hardening.
+HARDENING_CFLAGS  = -fstack-protector-strong -D_FORTIFY_SOURCE=2
```

Comment:
Hmm, this `-D_FORTIFY_SOURCE=2` might be problematic for ALT because there the default is `-D_FORTIFY_SOURCE=3`.

Reply:
I think it should be OK since per my reading of the changes here `HARDENING_CFLAGS` is prepended before the passed `CFLAGS`. Unless you have that as compiler default, not as part of default RPM `optflags`?

Reply:
Apparently, it's a compiler default, so now I see the following warnings:
```
<command-line>: warning: "_FORTIFY_SOURCE" redefined
<built-in>: note: this is the location of the previous definition
```

"info gcc" on that system displays the following note:
```
NOTE: In ALT 12.2.1-alt2 and later versions, '-D_FORTIFY_SOURCE=3' is set by default, and is activated when '-O' is set to 1 or higher.
```

Reply:
We'd probably avoid this specific issue by upgrading to `=3`, but overall it's not good we may override a potentially better default.